packages/utilities/gel-foundations/src/typography.js

```diff
@@ -1,11 +1,76 @@
-import { MEDIA_QUERY_TYPOGRAPHY } from './breakpoints';
+import { MEDIA_QUERY_TYPOGRAPHY } from '../breakpoints';
 
 const fontFamilyBase = ', Helvetica, Arial, sans-serif';
 
 export const GEL_FF_REITH_SERIF = `ReithSerif${fontFamilyBase}`;
 export const GEL_FF_REITH_SANS = `ReithSans${fontFamilyBase}`;
 export const GEL_FF_REITH_SANS_COND = `ReithSansCondensed${fontFamilyBase}`;
 
+function getTypeSizes(typogStyleName, importedScript) {
+  const typographyStyle = importedScript[typogStyleName];
+
+  const GROUP_A_FONT_SIZE_PX = typographyStyle.groupA.fontSize;
+  const GROUP_A_FONT_SIZE = `${GROUP_A_FONT_SIZE_PX / 16}rem`;
+  const GROUP_A_LINE_HEIGHT_PX = typographyStyle.groupA.lineHeight;
+  const GROUP_A_LINE_HEIGHT = `${GROUP_A_LINE_HEIGHT_PX / 16}rem`;
+
+  const GROUP_B_FONT_SIZE_PX = typographyStyle.groupB.fontSize;
+  const GROUP_B_FONT_SIZE = `${GROUP_B_FONT_SIZE_PX / 16}rem`;
+  const GROUP_B_LINE_HEIGHT_PX = typographyStyle.groupB.lineHeight;
+  const GROUP_B_LINE_HEIGHT = `${GROUP_B_LINE_HEIGHT_PX / 16}rem`;
+
+  const GROUP_D_FONT_SIZE_PX = typographyStyle.groupD.fontSize;
+  const GROUP_D_FONT_SIZE = `${GROUP_D_FONT_SIZE_PX / 16}rem`;
+  const GROUP_D_LINE_HEIGHT_PX = typographyStyle.groupD.lineHeight;
+  const GROUP_D_LINE_HEIGHT = `${GROUP_D_LINE_HEIGHT_PX / 16}rem`;
+
+  return `
+        font-size: ${GROUP_A_FONT_SIZE};
+        line-height: ${GROUP_A_LINE_HEIGHT};
+   
+        ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
+          font-size: ${GROUP_B_FONT_SIZE};
+          line-height: ${GROUP_B_LINE_HEIGHT};
+        }
+   
+        ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
+          font-size: ${GROUP_D_FONT_SIZE};
+          line-height: ${GROUP_D_LINE_HEIGHT};
+        }
+      `;
+}
+
+export const getCanon = script => getTypeSizes('canon', script);
+
+export const getTrafalgar = script => getTypeSizes('trafalgar', script);
+
+export const getParagon = script => getTypeSizes('paragon', script);
+
+export const getDoublePica = script => getTypeSizes('doublePica', script);
+
+export const getGreatPrimer = script => getTypeSizes('greatPrimer', script);
+
+export const getPica = script => getTypeSizes('pica', script);
+
+export const getLongPrimer = script => getTypeSizes('longPrimer', script);
+
+export const getBrevier = script => getTypeSizes('brevier', script);
+
+export const getMinion = script => getTypeSizes('minion', script);
+
+export const getAtlas = script => getTypeSizes('atlas', script);
+
+export const getElephant = script => getTypeSizes('elephant', script);
+
+export const getImperial = script => getTypeSizes('imperial', script);
+
+export const getRoyal = script => getTypeSizes('royal', script);
+
+export const getFoolscap = script => getTypeSizes('foolscap', script);
+
+export const getBodyCopy = script => getTypeSizes('bodyCopy', script);
+
+/* ORIGINAL CONSTANTS */
```

Comment:
What is the reason for making the GEL standard line heights and font sizes saved under 'news'? Can these not be under 'default', since these are what the gel-foundations typography settings should be if they don't need to be overridden.

Why is there a distinction between these typography settings and those in `latin.js`? I'd imagine they should be identical.


The reason our constant values for GEL typography are using `rem` instead of `px` values is for accessibility reasons - when we zoom into a page with `CTRL +` or `CMD +` or use a browser's zoom feature, using rem will ensure that the text scales well in conjuction with the rest of the page. 
[When we made this change, we wrote up a table of our type sizes here. ](https://github.com/bbc/simorgh/blob/67decf15656f785ae20caf0cef7e9e5657920766/src/app/lib/README.md#typography) 

It would be good to have all the scripts consistently using `rem`.

Reply:
1. Do you mean why we named it `news.js` instead of `default.js`? 

If not, please note that after Drew's yesterday suggestion about removing the dynamic script import in the `getTypeSizes` function because it won't work with webpack, now we will need to pass directly the script object with all the GEL Type Styles as an argument.

The script object will be imported from the Service Context and passed to the component via props, and as the `script` prop will be required we will need to always import either a NEWS or WS script.

**Import**
`import { arabicTypographyConfig } from '@bbc/gel-foundations/scripts';`
or
`import { newsTypographyConfig } from '@bbc/gel-foundations/scripts';`

**Usage**
`<Caption script={arabicTypographyConfig}>This is a caption.</Caption>,`

**Component declaration**
![Screenshot 2019-03-19 at 09 52 53](https://user-images.githubusercontent.com/46446236/54596613-ec00c280-4a2c-11e9-8995-a2385d469978.png)



If you have an altenative solution, could you please provide an example?

2. There is a distinction between `latin.js` and `news.js` because we assumed it would make it easier for UX to change something in WS and not in NEWS.

3. We are still using `rem`, then only difference is that we declare the GEL Types Styles using pixels in the scrips and then we convert them to `rem`, as explained in the previous comment.